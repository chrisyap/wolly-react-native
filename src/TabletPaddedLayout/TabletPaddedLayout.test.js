import React from 'react';
import PaddedLayoutView from './TabletPaddedLayout';
import { useDeviceOrientation } from '@react-native-community/hooks';

jest.mock('@react-native-community/hooks');

describe('PaddedLayoutView', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<PaddedLayoutView />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders in landscape correctly', () => {
    useDeviceOrientation.mockReturnValue({ landscape: true });
    const wrapper = shallow(<PaddedLayoutView />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with columns specified', () => {
    const props = {
      columns: 2,
    };
    const wrapper = shallow(<PaddedLayoutView {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
