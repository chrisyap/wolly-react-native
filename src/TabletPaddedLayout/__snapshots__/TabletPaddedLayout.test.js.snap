// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaddedLayoutView renders correctly 1`] = `
<Styled(View)
  mx="8.333333333333332%"
/>
`;

exports[`PaddedLayoutView renders correctly with columns specified 1`] = `
<Styled(View)
  mx="16.666666666666664%"
/>
`;

exports[`PaddedLayoutView renders in landscape correctly 1`] = `
<Styled(View)
  mx="16.666666666666664%"
/>
`;
